crates/vsock/src/vhu_vsock.rs

```diff
@@ -316,8 +317,13 @@ impl VhostUserBackend<VringRwLock, ()> for VhostUserVsockBackend {
             }
         }
 
-        if device_event != EVT_QUEUE_EVENT && thread.thread_backend.pending_rx() {
-            thread.process_rx(vring_rx, evt_idx)?;
+        if device_event != EVT_QUEUE_EVENT {
+            if thread.thread_backend.pending_rx() {
+                thread.process_rx(vring_rx, evt_idx)?;
+            }
+            if thread.thread_backend.pending_raw_pkts() {
+                thread.process_raw_pkts(vring_rx, evt_idx)?;
+            }
```

Comment:
Now it seems we call `process_rx` and `process_raw_pkts` (also the pending ones) always together, so might it be better to unify them to avoid future problems?

Effectively rx now depends on both backend and siblings.

Reply:
We still need the function to process raw packets only when a `SIBLING_EVENT` is received. So, instead I have modified the logic here to prevent the starvation of one queue by another.